mffpy/tests/test_writer.py

```diff
@@ -110,3 +111,58 @@ def test_writer_exports_JSON():
         remove(filename)
     except BaseException:
         raise AssertionError(f"""Clean-up failed of '{filename}'.""")
+
+##START##
+
+def test_streaming_writer_receives_bad_init_data():
+    """Test bin writer fails when initialized with non-int sampling rate"""
+    dirname = 'testdir.mff'
+    makedirs(dirname)
+    StreamingBinWriter(100, mffdir=dirname)
+    with pytest.raises(AssertionError):
+        StreamingBinWriter(100.0, mffdir=dirname)
+    rmtree(dirname)
+
+def test_streaming_writer_writes():
+    dirname = 'testdir2.mff'
+    # create some data and add it to a binary writer
+    device = 'HydroCel GSN 256 1.0'
+    num_samples = 10
+    num_channels = 256
+    sampling_rate = 128
+    # create an mffpy.Writer and add a file info, and the binary file
+    W = Writer(dirname)
+    W.create_directory()
+    b = StreamingBinWriter(sampling_rate=sampling_rate, data_type='EEG', mffdir=dirname)
+    data = np.random.randn(num_channels, num_samples).astype(np.float32)
+    b.add_block(data)
+    startdatetime = datetime.strptime(
+        '1984-02-18T14:00:10.000000+0100', XML._time_format)
+    W.addxml('fileInfo', recordTime=startdatetime)
+    W.add_coordinates_and_sensor_layout(device)
+    W.addbin(b)
+    W.write()
+    # read it again; compare the result
+    R = Reader(dirname)
+    assert R.startdatetime == startdatetime
+    # Read binary data and compare
+    read_data = R.get_physical_samples_from_epoch(R.epochs[0])
+    assert 'EEG' in read_data
+    read_data, t0 = read_data['EEG']
+    assert t0 == 0.0
+    assert read_data == pytest.approx(data)
+    layout = R.directory.filepointer('sensorLayout')
+    layout = XML.from_file(layout)
+    assert layout.name == device
+    # cleanup
+    try:
+        remove(join(dirname, 'info.xml'))
```

Comment:
Nit picking - but would `rmtree` on the `dirname` accomplish the same thing in a single line?

Reply:
Yeah. I had just copied the test case for the non-streaming and only changed those things I needed to. I can make this change to.